qdrant-landing/content/articles/serverless.md

```diff
@@ -0,0 +1,290 @@
+
+Now that you have the points inserted, you can search them by embedding:
+
+```rust
+use anyhow::Result;
+use qdrant_client::prelude::*;
+pub async fn search(
+    text: &str,
+    collection_name: String,
+    client: &Client,
+    api_key: &str,
+    qdrant: &QdrantClient,
+) -> Result<Vec<ScoredPoint>> {
+    Ok(qdrant.search_points(&SearchPoints {
+        collection_name,
+        limit: 5, // use what fits your use case here
+        with_payload: Some(true.into()),
+        vector: embed(client, text, api_key)?,
+        ..Default::default()
+    }).await?.result)
+}
+```
+
+And that's it. Obviously, you can also filter by adding a `filter: ...` field to the `SearchPoints`, and you will likely want to process the result further, but the example code already does that, so feel free to start from there in case you need this functionality.
+
+## Putting it all together
+
+Now that you have all the parts, it's time to join them up. Now copying and wiring up the snippets above is left as an exercise to the reader. Impatient minds can peruse the [example repo](https://github.com/qdrant/examples/tree/master/lambda-search) instead.
+
+You'll want to extend the `main` method a bit to connect with the Client once at the start, also get API keys from the environment so you don't need to compile them into the code. To do that, you can get them with `std::env::var(_)` from the rust code and set the environment from the AWS console.
+
+```bash
+$ aws lambda update-function-configuration \
+    --function-name $LAMBDA_FUNCTION_NAME \
+    --environment "Variables={QDRANT_URI=$QDRANT_URI,QDRANT_API_KEY=$QDRANT_API_KEY,OPENAI_API_KEY=$OPENAI_API_KEY}"`
+```
+
+In any event, you will arrive at one command line program to insert your data and one lambda function. The former can just be `cargo run` to set up the collection. For the latter, you can again call `cargo lambda` and the AWS console:
+
+```bash
+$ export LAMBDA_FUNCTION_NAME=search
+$ export LAMBDA_REGION=us-east-1
+$ cargo lambda build --release --arm --output-format zip
+  Downloaded libc v0.2.137
+# [..] output omitted for brevity
+    Finished release [optimized] target(s) in 1m 27s
+$ # Update the function
+$ aws lambda update-function-code --function-name $LAMBDA_FUNCTION_NAME \
+     --zip-file fileb://./target/lambda/page-search/bootstrap.zip \
+     --region $LAMBDA_REGION
+```
+
+## Discussion
+
+Lambda works by spinning up your function once the URL is called, so they don't need to keep the compute on hand unless it is actually used. This means that the first call will be burdened by some 1-2 seconds of latency for loading the function, later calls will resolve faster. Of course, there is also the latency for calling the embeddings provider and Qdrant. On the other hand, the free tier doesn't cost a thing, so you certainly get what you pay for. And for many use cases, a result within a few seconds is acceptable.
```

Comment:
seems like some of this should be at the top.....

i really think we need to push the free aspect - people love free